components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3823,9 +3836,11 @@ where
                         "target region is not found, drop proposals";
                         "region_id" => region_id
                     );
-                    for p in apply.cbs.drain(..) {
-                        let cmd = PendingCmd::<EK::Snapshot>::new(p.index, p.term, p.cb);
-                        notify_region_removed(apply.region_id, apply.peer_id, cmd);
+                    if !tikv_util::thread_group::is_shutdown(!cfg!(test)) {
+                        for p in apply.cbs.drain(..) {
+                            let cmd = PendingCmd::<EK::Snapshot>::new(p.index, p.term, p.cb);
+                            notify_region_removed(apply.region_id, apply.peer_id, cmd);
+                        }
```

Comment:
If a leader is removed by applying `confchange` (perhaps leader changed after the `confchange` proposed), invoking the callback while the raft log is not applied and `PeerFsm` still working may cause the same issue.

Reply:
First of all, a leader will not commit entries to delete itself, so the entry has to come from last term. And a leader will not propose new writes until it has applied to current term, so same issue will not happen.

Reply:
Oh, the `apply_res` will stop the `PeerFsm` first before the incoming write.